src/Network/Response.php

```diff
@@ -640,10 +645,17 @@ public function header($header = null, $value = null)
             if ($value === null) {
                 list($header, $value) = explode(':', $header, 2);
             }
-            $this->_headers[$header] = is_array($value) ? array_map('trim', $value) : trim($value);
+
+            if ($this->hasHeader($header)) {
+                $header = $this->headerNames[strtolower($header)];
+            } else {
+                $this->headerNames[strtolower($header)] = $header;
+            }
+
+            $this->headers[$header] = is_array($value) ? array_map('trim', $value) : trim($value);
         }
 
-        return $this->_headers;
+        return $this->headers;
```

Comment:
Won't this change the return value of `header()` with no arguments as the internal storage is different?